### PR TITLE
Build aos on RPI

### PR DESCRIPTION
**Pull request by mlohvynenko**

### CODE REVIEW

File: meta-aos-rpi-domd/recipes-core/images/aos-image-rpi.bb [hunk 1]
```diff
@@ -0,0 +1,69 @@
+SUMMARY = "Aos image for Raspberry Pi devices"
+
+require recipes-core/images/core-image-minimal.bb
+require recipes-core/images/aos-image.inc
+
+# Enable package manager
+EXTRA_IMAGE_FEATURES += "package-management"
```
Comment (al1img) [inline meta-aos-rpi-domd/recipes-core/images/aos-image-rpi.bb:7]: Why?

File: meta-aos-rpi-domd/recipes-core/images/aos-image-rpi.bb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+do_image[depends] += " \
+    rpi-bootfiles:do_deploy \
+    trusted-firmware-a:do_deploy \
+    ${@bb.utils.contains('RPI_USE_U_BOOT', '1', 'u-boot-tools-native:do_populate_sysroot', '',d)} \
+    ${@bb.utils.contains('RPI_USE_U_BOOT', '1', 'u-boot:do_deploy', '',d)} \
+    ${@bb.utils.contains('RPI_USE_U_BOOT', '1', 'u-boot-default-script:do_deploy', '',d)} \
+"
+
+PACKAGE_INSTALL:append = " \
```
Comment (al1img) [inline meta-aos-rpi-domd/recipes-core/images/aos-image-rpi.bb:19]: Unclear why it is needed?

File: meta-aos-rpi-domd/recipes-core/images/aos-image-rpi.bb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+IMAGE_INSTALL:append = " \
+    iproute2 \
+    iproute2-tc \
+    tcpdump \
+    e2fsprogs \
+    aos-messageproxy \
+"
+
+IMAGE_FSTYPES:remove = "wic.bz2 wic.bmap ext3"
```
Comment (al1img) [inline meta-aos-rpi-domd/recipes-core/images/aos-image-rpi.bb:63]: Should be set in yaml file to desired value

File: meta-aos-rpi-domd/recipes-aos/aos-iamanager/files/main-node/aos-vis-service.conf [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[Unit]
+Wants=aos-vis.service
+After=aos-vis.service
```
Comment (al1img) [inline meta-aos-rpi-domd/recipes-aos/aos-iamanager/files/main-node/aos-vis-service.conf:3]: New line at the end

File: meta-aos-rpi-domd/recipes-aos/aos-updatemanager/files/reboot-on-failure.conf [hunk 5]
```diff
@@ -0,0 +1,2 @@
+[Unit]
+OnFailure=aos-reboot.service
```
Comment (al1img) [inline meta-aos-rpi-domd/recipes-aos/aos-updatemanager/files/reboot-on-failure.conf:2]: New line at the end.

File: meta-aos-rpi-domd/recipes-aos/aos-updatemanager/files/secondary-node/aos_updatemanager.cfg [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                "SystemdChecker": {
+                    "SystemServices": [
+                        "aos-iamanager.service",
+                        "aos-servicemanager.service",
+                        "aos-messageproxy.service"
+                    ]
+                }
+            }
+        },
+        {
```
Comment (al1img) [inline meta-aos-rpi-domd/recipes-aos/aos-updatemanager/files/secondary-node/aos_updatemanager.cfg:27]: We don't support update boot for now. Remove it.

File: meta-aos-rpi-domd/recipes-core/images/aos-image-rpi.bb (hunk 2 above)
Comment (mlohvynenko) [reply meta-aos-rpi-domd/recipes-core/images/aos-image-rpi.bb]: Referred to xen team example.
Replaced with IMAGE_INSTALL, works fine.

File: meta-aos-rpi-domd/recipes-kernel/linux/files/aos.cfg [hunk 7]
```diff
@@ -0,0 +1,65 @@
+# Options required by Aos
+
+# Enable crypto device mapper to encrypt Aos partition
+CONFIG_BLK_DEV_DM_BUILTIN=y
```
Comment (al1img) [inline meta-aos-rpi-domd/recipes-kernel/linux/files/aos.cfg:4]: commit message: kernel options

File: aos-rpi.yaml [hunk 8]
```diff
@@ -30,11 +30,12 @@ variables:
   XT_XEN_DTB_NAME: "%{SOC_FAMILY}-%{MACHINE}-xen.dtb"
 
   # Addresses for u-boot
-  DOMD_IMG_ADDR: "0x1600000"
-  DOMD_DTB_ADDR: "0xb000000"
-  XEN_IMG_ADDR: "0xb100000"
-  XEN_DTBO_ADDR: "0xb300000"
-  XENPOLICY_IMG_ADDR: "0xb400000"
+  DOM0_IMG_ADDR: "0xe00000"  # 46 MB
```
Comment (al1img) [inline aos-rpi.yaml:33]: Need to check addresses.

**Review by al1img — APPROVED** [review]

**Pull request merged**